### PR TITLE
Update struct initializer docs

### PR DESCRIPTION
**Pull request by ntrel**
* Add member initializer example.
* `memberName:expression` can be used for non-static variables too.
* Move non-static struct initialization paragraph to dynamic initialization.
* Minor tweaks.

### CODE REVIEW

File: spec/struct.dd [hunk 1]
```diff
@@ -181,23 +189,12 @@ static S q = { 1, 2 }; // q.a = 1, q.b = 2, q.c = 0, q.d = 7
 ------
 
         $(P Struct literals can also be used to initialize statics, but
-        they must be evaluable at compile time.)
+        they must be evaluatable at compile time.)
```
Comment (ntrel) [inline spec/struct.dd]: Changed for consistency with other pages on dlang.org.

Comment (andralex) [inline spec/struct.dd]: We need to replace "evaluatable" (which is not a word) with "evaluable" (which is) throughout our docs:

https://www.merriam-webster.com/dictionary/evaluatable
https://www.merriam-webster.com/dictionary/evaluable

**Comment by JohanEngelen** [comment]
Thanks.
What I am missing (what trigger my forum post) is explicit mention of these two kinds of examples:

```d
static S x = {1,  2}; // <-- member:value  syntax is not the only way for statics
S y = { a:1 }; // <-- member:value is also allowed for non-statics
```

File: spec/struct.dd (hunk 1 above)
Comment (WalterBright) [inline spec/struct.dd]: A PR should not insert gratuitous errors to be fixed in a later PR, regardless of whether other pages do it. Please fix.

Comment (WalterBright) [inline spec/struct.dd]: And the other instances in the context lines here. These lines are being changed anyway.

Comment (WalterBright) [inline spec/struct.dd]: The rest is good to go.

**Comment by ntrel** [comment]
@JohanEngelen C-style static initialization `static S q = { 1, 2 };` was already in the spec. The last commit now has `S q = { 1, b:i };`.

@WalterBright Fixed using *evaluable*. I also added an example showing that C-style initialization and named member initialization can be combined. Also I reduced an uncalled `S.opCall(S)` in a later example to just `assert(0);` for clarity (this is part of the struct initialization docs too).

**Comment by JohanEngelen** [comment]
Nice.

**Comment by ntrel** [comment]
@wilzbach great, thanks 😊

**Pull request merged**